src/modules/cpu/kernel/fog.hpp

```diff
@@ -0,0 +1,1524 @@
+            std::uniform_int_distribution<> distribY(0, fog_height - roi.xywhROI.roiHeight);
+
+            RppiPoint maskLoc;
+            maskLoc.x = distribX(gen);
+            maskLoc.y = distribY(gen);
+
+            fogAlphaMaskPtr = &fogAlphaMask_1280_720[(fog_width * maskLoc.y) + maskLoc.x];
+            fogIntensityMaskPtr = &fogIntensityMask_1280_720[(fog_width * maskLoc.y) + maskLoc.x];
+
+        }
+        else
+        {
+            fog_width = 1920;
+            fog_height = 1080;
+            std::random_device rd;  // Random number engine seed
+            std::mt19937 gen(rd()); // Seeding rd() to fast mersenne twister engine
+            std::uniform_int_distribution<> distribX(0, fog_width - roi.xywhROI.roiWidth);
+            std::uniform_int_distribution<> distribY(0, fog_height - roi.xywhROI.roiHeight);
+
+            RppiPoint maskLoc;
+            maskLoc.x = distribX(gen);
+            maskLoc.y = distribY(gen);
+
+            fogAlphaMaskPtr = &fogAlphaMask_1920_1080[(fog_width * maskLoc.y) + maskLoc.x];
+            fogIntensityMaskPtr = &fogIntensityMask_1920_1080[(fog_width * maskLoc.y) + maskLoc.x];
+
+        }
+
+
+        Rpp32u alignedLength = (bufferLength / 24) * 24;
+        Rpp32u vectorIncrement = 24;
+        Rpp32u vectorIncrementPerChannel = 8;
+
+        // Fog without fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp16f *srcPtrRow, *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            Rpp32f *fogAlphaMaskPtrRow, *fogIntensityMaskPtrRow;
+            srcPtrRow = srcPtrChannel;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+            fogAlphaMaskPtrRow = fogAlphaMaskPtr;
+            fogIntensityMaskPtrRow = fogIntensityMaskPtr;
+
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp16f *srcPtrTemp, *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                Rpp32f *fogAlphaMaskPtrTemp, *fogIntensityMaskPtrTemp;
+                srcPtrTemp = srcPtrRow;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+                fogAlphaMaskPtrTemp = fogAlphaMaskPtrRow;
+                fogIntensityMaskPtrTemp = fogIntensityMaskPtrRow;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                {
+                    Rpp32f srcPtrTemp_ps[24];
+                    Rpp32f dstPtrTempR_ps[8], dstPtrTempG_ps[8], dstPtrTempB_ps[8];
+                    for(int cnt = 0; cnt < vectorIncrement; cnt++)
+                        srcPtrTemp_ps[cnt] = (Rpp32f) srcPtrTemp[cnt];
```

Comment:
Lets not try to follow this old approach of copying F16 buffers to F32 raw buffers
Lets try to load directly from F16 buffers using instrinics
```
__m256 p;
p = _mm256_cvtph_ps(_mm_castps_si128(_mm_loadu_ps(reinterpret_cast<Rpp32f *>(srcPtrTemp))));
```
This will load 8 F16 values and converts to 8 F32 values directly
You can create some load functions for F16 that will load using intrinsics and convert to F32